apis/Google.Cloud.Firestore/Google.Cloud.Firestore/ValueSerializer.cs

```diff
@@ -124,8 +128,13 @@ internal static Value Serialize(object value)
             GaxPreconditions.CheckNotNull(value, nameof(value));
             switch (value)
```

Comment:
Perhaps not for this PR, but the four cases here are pretty much identical to those above (except the result is then passed to `CreateMapValue` above). Do you think it would make sense to extract them into some kind of `TrySerializeMap` helper?

Reply:
Potentially, yes. At some point I'd like to redo a lot of this code to use expression trees or something similar - we can make it a lot faster, I suspect. But yes, a little refactoring before then may be a good idea.